src/test/java/de/mediathekview/mserver/crawler/wdr/parser/WdrVideoJsonDeserializerTest.java

```diff
@@ -0,0 +1,96 @@
+        Optional.empty(),
+        Optional.empty()
+      },
+      {
+        "/wdr/wdr_video3.json",
+        Optional.of("http://wdrmedien-a.akamaihd.net/medp/ondemand/weltweit/fsk0/52/528067/528067_5542163.xml"),
+        "http://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/52/528067/,528067_5540994,528067_5540993,528067_5540992,528067_5540996,528067_5540995,.mp4.csmil/master.m3u8",
+        Optional.of("http://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/52/528067/,528067_5542160,528067_5542162,528067_5542161,.mp4.csmil/master.m3u8"),
+        Optional.empty()
+      },
+      {
+        "/wdr/wdr_video_v1_1.json",
+        Optional.empty(),
+        "http://ondemand-ww.wdr.de/medp/fsk0/47/476693/476693_12040646.mp4",
+        Optional.empty(),
+        Optional.empty()
+      },
+      {
+        "/wdr/wdr_video_with_ad_dgs.json",
+        Optional.of("http://wdrmedien-a.akamaihd.net/medp/ondemand/weltweit/fsk0/158/1583693/1583693_18232422.xml"),
+        "http://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/158/1583693/,1583693_18232361,1583693_18232358,1583693_18232363,1583693_18232362,1583693_18232359,1583693_18232360,.mp4.csmil/master.m3u8",
+        Optional.of("http://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/158/1583693/,1583693_18232391,1583693_18232389,1583693_18232393,1583693_18232392,1583693_18232388,1583693_18232390,.mp4.csmil/master.m3u8"),
+        Optional.of("http://wdradaptiv-vh.akamaihd.net/i/medp/ondemand/weltweit/fsk0/158/1583693/,1583693_18232344,1583693_18232347,1583693_18232346,1583693_18232345,1583693_18232343,1583693_18232342,.mp4.csmil/master.m3u8")
+      }
+    });
+  }
+
+  private final String jsonFile;
+  private final Optional<String> expectedSubtitle;
+  private final String expectedM3U8Url;
+  private final Optional<String> expectedSignLanguageUrl;
+  private final Optional<String> expectedAudioDescriptionUrl;
+  
+  public WdrVideoJsonDeserializerTest(final String aJsonFile, 
+          final Optional<String> aExpectedSubtitle, 
+          final String aExpectedM3U8Url,
+          final Optional<String> aExpectedSignLanguageUrl,
+          final Optional<String> aExpectedAudioDescriptionUrl) {
+    
+    jsonFile = aJsonFile;
+    expectedSubtitle = aExpectedSubtitle;
+    expectedM3U8Url = aExpectedM3U8Url;
+    expectedSignLanguageUrl = aExpectedSignLanguageUrl;
+    expectedAudioDescriptionUrl = aExpectedAudioDescriptionUrl;
+  }
+  
+  @Test
+  public void test() {
+    final JsonElement jsonElement = JsonFileReader.readJson(jsonFile);
+    
+    WdrVideoJsonDeserializer target = new WdrVideoJsonDeserializer("http:");
+    Optional<WdrMediaDto> actual = target.deserialize(jsonElement, null, null);
+    
+    assertThat(actual, notNullValue());
+    assertThat(actual.isPresent(), equalTo(true));
+    
+    WdrMediaDto actualDto = actual.get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "actual.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)